src/garage/envs/environment.py

```diff
@@ -0,0 +1,189 @@
+"""Base Garage Environment API."""
+
+import abc
+
+
+class Environment(abc.ABC):
```

Comment:
how does it look if you, for instance, try to rewrite DmControlEnv to match this interface?

Reply:
i think you should do this exercise before merging this PR. We can only get so far speculating how to design an API -- the real proof is trying to implement it and use it.

Reply:
Please see my latest commit. The code only serves as a code snippet example as changes from TimeStep are not included in the PR. 

This is a great exercise as I realized it makes more sense to let the environment know `max_n_steps` so it can produce TIMEOUT time step directly. 

Pls let me know your thoughts;)

Reply:
looks great! can you include a rewrite of PointEnv and/or GridWorldEnv?